tony-core/src/main/java/com/linkedin/tony/tensorflow/TonySession.java

```diff
@@ -362,10 +362,8 @@ private TonyTask getTask(String jobName, String taskIndex) {
     return null;
   }
 
-  private boolean isChief(String jobName, String jobIndex) {
-    String chiefName = tonyConf.get(TonyConfigurationKeys.CHIEF_NAME, TonyConfigurationKeys.DEFAULT_CHIEF_NAME);
-    String chiefIndex = tonyConf.get(TonyConfigurationKeys.CHIEF_INDEX, TonyConfigurationKeys.DEFAULT_CHIEF_INDEX);
-    return jobName.equals(chiefName) && jobIndex.equals(chiefIndex);
+  private boolean isChief(String jobName) {
```

Comment:
this part is not right. By default, worker 0 is chief worker, and we have logic that if chief is dead, we stop training (fix the hanging training scenario when chief worker is head and other workers are waiting)

Reply:
I see what you're getting at. I think we need to modify the `TestTonyE2E.testWaitingForChiefToFinish` test, since it still passed after my changes :).

Reply:
I added back the worker 0 check. However, I don't think the `TestTonyE2E.testWaitingForChiefToFinish` test was actually testing short-circuiting the training if the chief worker fails, so I removed it.